src/mbgl/renderer/render_source.hpp

```diff
@@ -118,6 +118,10 @@ class RenderSource : protected TileObserver {
 
     void onTileChanged(Tile&) override;
     void onTileError(Tile&, std::exception_ptr) final;
+    void onIdealTilesChanged(const std::vector<UnwrappedTileID>&) final;
+
+private:
+    std::unordered_map<UnwrappedTileID, bool> idealTilesLoaded;
```

Comment:
instead, it could be `std::set<UnwrappedTileID> idealTilesLoaded;` that would hold only loaded tiles

Reply:
Done